### ISSUE TITLE
Kubectl run hangs on 5-6 minutes. 

### ISSUE THREAD

**Issue by patotskiy** [kubernetes/kubernetes#105554]
I'm trying to deploy image to Kubernetes cluster using following command: 
`kubectl run -n dev --rm api-tests --image=fo/api-tests:42276-simple_proj  -ojson --attach=true --restart=Never`

After execution pod hangs in 2 different statuses: `Completed` or `Error`. 
After 5-6 mins I receive following log: 
```
pod "api-tests" deleted
pod dev/api-tests terminated (Error)
```
It is correct behaviour, but how to make it so that the execution does not freeze for 5-6 minutes. 

Kubectl Client Version: v1.22.2
Kubectl Server Version: v1.19.13

We need:
1. Execute pod once 
2. Return exit code of this pod

/sig cli

**Comment by neolit123** [comment]
> Kubectl Client Version: v1.22.2
> Kubectl Server Version: v1.19.13

please note that this is not a supported skew, see:
https://kubernetes.io/releases/version-skew-policy/#kubectl

**Comment by brianpursley** [comment]
This is caused by the same thing as https://github.com/kubernetes/kubectl/issues/1098

The bug was fixed by https://github.com/kubernetes/kubernetes/pull/105384 and will be included in 1.22.3.

The workaround for now is to disable command headers using `KUBECTL_COMMAND_HEADERS=false`.  Command headers are informational for the cluster, so it will not affect functionality at all to disable them.

Example:

```
KUBECTL_COMMAND_HEADERS=false kubectl run -n dev --rm foo --image=busybox -o json --attach=true --restart=Never -- sleep 5; echo "Success!"
```

Alternatively you can use kubectl version 1.21.x or earlier.

**Comment by eddiezane** [comment]
This should be fixed now that 1.22.3 shipped.

/close
